### DISCUSSION TITLE
Invalid project-package combination for docfx 2.75.3

### DISCUSSION THREAD

**Discussion by omidsotooni** [dotnet/docfx#9779] [Q&A] [answered]
Hi, i want install docfx on my project, target framework is net8.0 but get this error:

```

Restoring packages for C:\Users\Omid\Documents\Projects\WebApiTest.csproj...
  CACHE https://api.nuget.org/v3/vulnerabilities/index.json
  CACHE https://api.nuget.org/v3-vulnerabilities/2024.03.07.05.31.04/vulnerability.base.json
  CACHE https://api.nuget.org/v3-vulnerabilities/2024.03.07.05.31.04/2024.03.10.23.31.34/vulnerability.update.json
NU1202: Package docfx 2.75.3 is not compatible with net8.0 (.NETCoreApp,Version=v8.0). Package docfx 2.75.3 supports:
  - net6.0 (.NETCoreApp,Version=v6.0) / any
  - net7.0 (.NETCoreApp,Version=v7.0) / any
  - net8.0 (.NETCoreApp,Version=v8.0) / any
  - playwright (.playwright,Version=v0.0) / node
  - playwright (.playwright,Version=v0.0) / package
NU1212: Invalid project-package combination for docfx 2.75.3. DotnetToolReference project style can only contain references of the DotnetTool type 
Package 'docfx 2.75.3' has a package type 'DotnetTool' that is not supported by project 'src\DotNetCoreProjects.APIs\WebApiTest'.
Time Elapsed: 00:00:00.4845976
========== Finished ==========

```

In documents wrote support 6.x, 7.x and 8.x.

**Comment by filzrev** [comment]
> NU1212: Invalid project-package combination for docfx 2.75.3. DotnetToolReference project style can only contain references of the DotnetTool type 
> Package 'docfx 2.75.3' has a package type 'DotnetTool' that is not supported by project 'src\DotNetCoreProjects.APIs\WebApiTest'.

`docfx` NuGet package is prepared for .NET tools installation. It can't be reference by using `<PackageReference>`.

If you want to use docfx functionality from .NET projects.
It need to reference `Docfx.App` package instead,

**Reply by twalters-inovalon** [reply]
I’m getting the same error when trying to add the docfx 2.75.3 NuGet package to a .net6 library project in my solution and am not sure I understand this answer. Are you saying docfx must be installed locally?  I used to just add it to a project in the solution – there was no need to install locally.  Thanks.

**Reply by filzrev** [reply, marked as answer]
`docfx` v2.60 or later package is released as [.NET Tools](https://learn.microsoft.com/en-us/dotnet/core/tools/global-tools) package.
So it's intended to be installed by `dotnet tool` command. 
and it can't reference from projects.

What's the purpose of referencing `docfx` package from projects?
If you are intended `docfx.console` package's MSBuild integration.
It's no longer supported.

**Reply by twalters-inovalon** [reply]
Thanks.  When installed as a NuGet on a project, there was no need to install as a tool locally, so there was no dependency on having a tool installed to work on the solution.  Now each developer will need to install the tool.
